### PR TITLE
Cleanup code and tests in Polyrepo tests

### LINKED ISSUE #1079: Cleanup Polyrepo test code
There are various places where this code could be cleaned up. The main goal of this issue is to reduce unneeded verbosity/duplication/redundancy where possible.

We need a different framework than what's currently used in the code to
1. add various things to the package's configuration
2. make packages that 
    1. have unique src and test module names
    2. can be depended on in other packages' code

See https://github.com/purescript/spago/pull/1075#discussion_r1357889757 for context.

### PR DESCRIPTION
**Pull request by JordanMartinez**
### Description of the change

Fixes #1079 

### Checklist:

- [ ] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

File: test/Spago/Build/Polyrepo.purs [hunk 1]
```diff
@@ -68,97 +71,189 @@ spec = Spec.describe "polyrepo" do
...
-              <>
-                ( packageName # maybe [] \pkgName ->
-                    [ ""
-                    , "packageName :: String"
-                    , "packageName = \"" <> pkgName <> "\""
-                    ]
-                )
-        }
+    mkDependencies :: Array String -> Config.Dependencies
+    mkDependencies = Config.Dependencies <<< Map.fromFoldable <<< map (flip Tuple Nothing <<< mkPackageName)
```
Comment (JordanMartinez) [inline test/Spago/Build/Polyrepo.purs:75]: Maybe this should go in `Test/Prelude.purs`?

File: test/Spago/Build/Polyrepo.purs [hunk 2]
```diff
@@ -68,97 +71,189 @@ spec = Spec.describe "polyrepo" do
...
+      Init.defaultConfig' $ PackageOnly $ snd finalConfig
+
+    configAddSrcStrict :: Tuple String Init.DefaultConfigPackageOptions -> Tuple String Init.DefaultConfigPackageOptions
+    configAddSrcStrict = map \r -> r
+      { build = Just
+          { strict: Just true
+          , censorProjectWarnings: r.build >>= _.censorProjectWarnings
+          , pedanticPackages: r.build >>= _.pedanticPackages
+          }
+      }
```
Comment (JordanMartinez) [inline test/Spago/Build/Polyrepo.purs:101]: Should these functions go in `Init/Config.purs`?

File: test/Spago/Build/Polyrepo.purs (hunk 1 above)
Comment (f-f) [reply test/Spago/Build/Polyrepo.purs:75]: Absolutely!

File: test/Spago/Build/Polyrepo.purs (hunk 2 above)
Comment (f-f) [reply test/Spago/Build/Polyrepo.purs:101]: We are using these in tests only so I don't think they should be in the project source tree; at best we might be able to reuse them in other tests outside of the polyrepo stuff, so we can move them to the Test.Prelude or some other common module if we find refactoring opportunities in other tests

**Review by f-f — APPROVED** [review]
This is nice 👏

**Pull request merged**